pkg/network/infraconfigurators/bridge_test.go

```diff
@@ -0,0 +1,192 @@
+package infraconfigurators
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/vishvananda/netlink"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	netdriver "kubevirt.io/kubevirt/pkg/network/driver"
+)
+
+type Option func(handler *netdriver.MockNetworkHandler)
+
+func newMockedBridgeConfigurator(
+	vmi *v1.VirtualMachineInstance,
+	iface *v1.Interface,
+	handler *netdriver.MockNetworkHandler,
+	bridgeIfaceName string,
+	launcherPID int,
+	options ...Option) *BridgePodNetworkConfigurator {
+	configurator := NewBridgePodNetworkConfigurator(vmi, iface, bridgeIfaceName, launcherPID, handler)
+	for _, option := range options {
+		option(handler)
+	}
+	return configurator
+}
+
+var _ = Describe("Bridge infrastructure configurator", func() {
+	var (
+		ctrl    *gomock.Controller
+		handler *netdriver.MockNetworkHandler
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		handler = netdriver.NewMockNetworkHandler(ctrl)
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	const (
+		bridgeIfaceName = "br0"
+	)
+
+	newVMIWithBridgeInterface := func(namespace string, name string) *v1.VirtualMachineInstance {
```

Comment:
Please take out all lambda function from the test containers to the bottom of the file, so I will not need to delve in the details. I want to read the tests and what they do first, then look into the details if I care.
I do not see any advantage in putting these details in front of the tests or between them. We do not write production code this way, and we should not do it in tests IMO.